libs/server/Resp/RespServerSessionSlotVerify.cs

```diff
@@ -30,24 +31,99 @@ bool NetworkSingleKeySlotVerify(ReadOnlySpan<byte> key, bool readOnly)
         bool NetworkKeyArraySlotVerify(Span<ArgSlice> keys, bool readOnly, int count = -1)
             => clusterSession != null && clusterSession.NetworkKeyArraySlotVerify(keys, readOnly, SessionAsking, ref dcurr, ref dend, count);
 
-        /// <summary>
-        /// Verify if the corresponding command can be served given the status of the slot associated with the parsed keys.
-        /// </summary>
-        /// <param name="readOnly"></param>
-        /// <param name="firstKey"></param>
-        /// <param name="lastKey"></param>
-        /// <param name="step"></param>
-        /// <returns></returns>
-        bool NetworkMultiKeySlotVerify(bool readOnly = false, int firstKey = 0, int lastKey = -1, int step = 1)
+        bool CanServeSlot(RespCommand cmd)
         {
+            // If cluster is disable all commands
             if (clusterSession == null)
+                return true;
+
+            // Verify slot for command if it falls into data command category
+            if (!cmd.IsDataCommand())
+                return true;
+
+            cmd = cmd.NormalizeForACLs();
+            if (!RespCommandsInfo.TryFastGetRespCommandInfo(cmd, out var commandInfo))
+                // This only happens if we failed to parse the json file
                 return false;
-            csvi.readOnly = readOnly;
+
+            // The provided command is not a data command
+            // so we can serve without any slot restrictions
+            if (commandInfo == null)
+                return true;
+
+            var specs = commandInfo.KeySpecifications;
```

Comment:
Might be neater to have GetFirstKey/GetLastKey/GetKeyStep abstract methods as part of FindKeysKeySpecMethodBase and implement them for each FindKeys type. WDYT?

Reply:
We could do that, but it will require evaluating the switch statement multiple times per Get call. If you want to move it towards the FindKeysBase it might be best to have GetSlotVerificationInput instead of having different calls.